### PR TITLE
[#102] [#103] [UI] As a user, I can see the text field question and text area question

### LINKED ISSUE #32: [Integration] As a user, I can see the questions in the survey and answer them
## Why

After choosing the `Start Survey` button on the survey detail screen, the user can load all the questions in the survey, and start answering them.

## Acceptance criteria

After choosing the `Start Survey` button on the survey detail screen, send an API request to get the question list.

Follow the [API document for /api/v1/surveys/{survey-id}`](https://documenter.getpostman.com/view/11835486/T1LJjTqj#142b8597-e176-4e74-95a5-cb48ed5115f0).

- If the response returns with success, navigate to the first question screen.

- If the response returns with failure, show the error alert with a message.

The user answers the question and goes to the next question by tapping on the next button at the bottom right corner of the question screen.

The user cannot go back to the previous question.

## Resources

- [API document for /api/v1/surveys/{survey-id}`](https://documenter.getpostman.com/view/11835486/T1LJjTqj#142b8597-e176-4e74-95a5-cb48ed5115f0)

### LINKED ISSUE #102: [UI] As a user, I can see the text field question
## Why

The user can see the text field questions and input answers.

## Acceptance criteria

Show text fields and their placeholder

### LINKED ISSUE #103: [UI] As a user, I can see the text area question
## Why

The user can see the text area question and input answer.

## Acceptance criteria

Show text area and its placeholder.

### PR DESCRIPTION
**Pull request by nmint8m**
- Close #102 #103 #32

## What happened 👀

The user can see the text field questions and text area question
The user can input answers.

## Insight 📝

N/A

### CODE REVIEW

File: lib/views/answer/form/form_with_text_field_view.dart [hunk 1]
```diff
@@ -1,10 +1,82 @@
...
-    return Container();
+    return _buildForm();
+  }
+
+  Widget _buildForm() {
+    return Column(
+      mainAxisAlignment: MainAxisAlignment.center,
+      children: widget.uiModels.map(
+        (uiModel) {
+          final controller = TextEditingController();
```
Comment (manh-t) [inline lib/views/answer/form/form_with_text_field_view.dart]: technically, we should create the controllers once in StatefulWidget. So putting the initialization in the `build` method will trigger the controller initialization multiple times. Can you put this in `initState` instead?

File: lib/views/answer/form/form_with_text_field_view.dart [hunk 2]
```diff
@@ -1,10 +1,82 @@
...
+          answers[uiModel.id] = '';
+          controller.addListener(
+            () {
+              answers[uiModel.id] = controller.text;
+              widget.onChange(answers);
+            },
+          );
+          _controllers.add(controller);
```
Comment (manh-t) [inline lib/views/answer/form/form_with_text_field_view.dart]: same for the listeners

**Review by manh-t — CHANGES_REQUESTED** [review]

File: lib/views/answer/form/form_with_text_field_view.dart [hunk 3]
```diff
@@ -1,10 +1,96 @@
...
+    for (var controller in _controllers) {
+      controller.dispose();
+    }
+    super.dispose();
+  }
 
   @override
   Widget build(BuildContext context) {
-    return Container();
+    _buildControllers();
```
Comment (manh-t) [inline lib/views/answer/form/form_with_text_field_view.dart]: @nmint8m still, this method is in `build` 😓 Can't we set it in `initState`?
```
@override
  void initState() {
    // TODO: implement initState
    super.initState();
    Future.delayed(Duration(seconds: 0), () {
      _buildControllers();
    });
  }
```

**Comment by nmint8m** [comment]
@manh-t Sure, I moved the code!

**Review by manh-t — APPROVED** [review]

**Pull request merged**